app/models/git/repository.rb

```diff
@@ -67,6 +67,13 @@ def lookup_commit(sha, update_on_failure: true)
       lookup_commit(sha, update_on_failure: false)
     end
 
+    def file_exists?(commit, path)
+      commit.tree.path(path)
+      true
```

Comment:
```suggestion
      !!commit.tree.path(path)
```

(Presuming the result is a string)

Reply:
I was still testing stuff here :)